File: 백준/Silver/10828. 스택/스택.py
# N회를 입력 후에 각 기능을 입력하여 결과를 출력한다
# input 보다 sys 처리속도가 더 빠르다
# 저장을 위해 빈 리스트의 stack이 필요하다
# word를 입력할 새로운 입력창도 만들자

import sys
input = sys.stdin.readline
N = int(input())
stack = []

for i in range(N):
    word = input().split()
    if word[0] == 'push':
        stack.append(word[1])
    elif word[0] == 'pop':
        if not stack:
            print(-1)
        else:
            print(stack.pop(-1))
    elif word[0] == 'size':
        print(len(stack))
    elif word[0] == 'empty':
        if not stack:
            print(1)
        else:
            print(0)
    elif word[0] == 'top':
        if not stack:
            print(-1)
        else:
            print(stack[-1])

File: 백준/Silver/10828. 스택/스택.java
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.BufferedReader;

public class Main {
	static int[] stack = new int[10001];
	static int idx = 0;
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		for (int i = 0; i < N; i++) {
			String s = br.readLine();
			my_stack(s);
		}
	}
	
	
	
	public static void my_stack(String commands) {
		if (commands.startsWith("push")) {
			int num = Integer.parseInt(commands.split(" ")[1]);
			stack[idx] = num;
			++idx;
		}
		else if (commands.equals("pop")) {
			if (idx == 0) System.out.println(-1);
			else {
				--idx;
				System.out.println(stack[idx]);
			}
		}
		else if (commands.equals("size")) System.out.println(idx);
		else if (commands.equals("top")) {
			if (idx == 0) System.out.println(-1);
			else System.out.println(stack[idx-1]);
		}
		else if (commands.equals("empty")) {
			if (idx == 0) System.out.println(1);
			else System.out.println(0);
		} 
	
	
	}
	
	
	
	
	
}